core/src/env_platform/db_handler.bal

```diff
@@ -171,3 +171,97 @@ function updateApplication(TreeRemovalForm form, string applicationId) returns b
+# 
+# + userId - Id of the user.
+# + return - This function will return either user is valid or 
+# error if there is a mongodb:DatabaseError.
+function isValidUser(string userId) returns boolean|error {
+
+    int countDocuments = check usersCollection->countDocuments({id: userId});
+    if (countDocuments == 1) {
+        return true;
+    } else if (countDocuments == 0) {
+        return false;
+    } else {
+        return error("Issue having duplicate user Ids", message = "There are two or more similar user Ids in the system");
+    }
+}
+
+# The `saveApplicationInUser` function will save the application reference in the corresponding user.
+# 
+# + userId - Id of the user.
+# + applicationId - Id of the application saved.
+# + applicationType - Type of the application.
+# + return - This function will return either application is saved in the given user document or 
+# error if there is a mongodb:DatabaseError.
+function saveApplicationInUser(string userId, string applicationId, string applicationType) returns boolean|error {
+
+    boolean isValid = check isValidUser(userId);
+    if (isValid) {
+
+        // Get the user information.
+        map<json>[] find = check usersCollection->find({id: userId});
+        json|error applications = find[0].applications;
+        log:printDebug("User applications are " + applications.toString());
+
+        // Construct the applicationList.
+        json[] applicationList;
+        if (applications is error) {
+            applicationList = [{id: applicationId, name: applicationType}];
+        } else {
+            applicationList = <json[]>applications;
+            applicationList.push(<json>{id: applicationId, name: applicationType});
+        }
+
+        // Update the user applications array with incoming value.
+        int updated = check usersCollection->update({"applications": applicationList}, {id: userId});
+        log:printDebug("Updated application list is: " + applicationList.toString());
+        if (updated > 0) {
+            return true;
+        } else {
+            return false;
+        }
+    } else {
+        return error("Invalid user", message = "Couldn't find the user with given userId");
+    }
+}
+
+# The `removeApplicationInUser` function will remove the application reference in the corresponding user.
+# 
+# + userId - Id of the user.
+# + applicationId - Id of the application which should be removed.
+# + return - This function will return either application is removed in the given user document or 
+# error if there is a mongodb:DatabaseError.
+function removeApplicationInUser(string userId, string applicationId) returns boolean|error {
+    boolean isValid = check isValidUser(userId);
+    if (isValid) {
+
+        // Get the user information.
+        map<json>[] find = check usersCollection->find({id: userId});
+        json|error applications = find[0].applications;
+        log:printDebug("User applications are " + applications.toString());
```

Comment:
```suggestion
        log:printDebug("The user's applications are: " + applications.toString());
```

Reply:
Little worried how the debug message will show up. Might not be developer friendly. Could split the applications.toString() part into a different debug log

Reply:
Should I make a for loop to show up each and every application detail in separate lines.

Reply:
Not necessary in the initial iteration, but could be added as a TODO.

Reply:
Yeah but I will resolve it.